### ISSUE TITLE
No access to original HTTP request in GET request

### ISSUE THREAD

**Issue by frosenberg** [go-swagger/go-swagger#550]
## Problem statement

The generated handler function for a request without parameters does not allow one to access the original HTTP request because it does not have a `params` object in the signature. For the swagger snippet below I see the following handler being generated in the `configure_whatever.go`:

```
api.HealthGetHealthHandler = health.GetHealthHandlerFunc(func() middleware.Responder {
  return middleware.NotImplemented("operation health.GetHealth has not yet been implemented")
})
```

There is no argument in the signature that allows me to access the original HTTP request to, for example, read an HTTP header. 

## Swagger specification

```
swagger: '2.0'
info:
  title: Swagger issue
  description: foobar
  version: "1.0.0"
host: example.com
schemes:
  - http
basePath: /v1
consumes:
  - application/json
produces:
  - application/json

paths:
  /health:
    get:
      summary: Health check resource
      description: Checks the health of the API.
      tags:
        - Health
      responses:
        200:
          description: Service is healty
          schema:
            $ref: '#/definitions/Health'
        503:
          description: Service not available

definitions:
  Health:
    type: object
    properties:
      status:
        type: string
        description: Short status message about the health.

```

## Steps to reproduce

1. Generate server
```
swagger generate server -f swagger.yml
```

2. Look at the handler and try go access original request.

**Comment by aocole** [comment]
I am having the same problem. @frosenberg, were you able to work around this?

**Comment by casualjim** [comment]
Can somebody confirm this fixes it? https://github.com/go-swagger/go-swagger/tree/550-always-generate-params

**Comment by frosenberg** [comment]
@aocole what I did was to declare the headers I needed as optional parameters at the top level like this:
```
# Optional header parameters
parameters:
  DlaasGroup:
      name: X-DLaaS-Roles
      description: Roles of a user
      type: string
      in: header
      required: false
```
Then in my resources I just reference them:

```
paths:
  /tickets:
    parameters:
      - $ref: '#/parameters/DlaasGroup'
    post:
      summary: ...
      description: ...
```

This way the client and server have explicit methods to set/get them. Not ideal but it works. Since the header are coming from our proxy the client wont need to set them and I would have preferred to handle transparently for the user but that is an OK solution for now. @casualjim's fix looks like it should do exactly what I need but I have not had time to look at it.

**Comment by aocole** [comment]
That's a good workaround @frosenberg. I am going to backport @casualjim's fix from #558 in to a fork until I can upgrade after #467
